src/utils/activity-object-helper.ts

```diff
@@ -0,0 +1,114 @@
+import { ActivityObject, Link, Image } from "activitypub-objects";
+import { RdfLangString } from "@/model/rdf-lang-string";
+import { Activity } from "@/model/mitra-activity";
+
+export class ActivityObjectHelper {
+  public static hasProperty(obj: object, property: string): boolean {
+    try {
+      return property in obj;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  public static extractActorName(
+    object: ActivityObject | Link | URL | Array<ActivityObject | URL>
+  ): string | undefined {
+    const lang = navigator.language.substr(0, 2);
+    if (ActivityObjectHelper.hasProperty(object, "name")) {
+      return (object as ActivityObject).name;
+    } else if (ActivityObjectHelper.hasProperty(object, "nameMap")) {
+      return (object as RdfLangString).nameMap[lang];
+    } else if (ActivityObjectHelper.hasProperty(object, "map")) {
+      return (object as RdfLangString).map[lang];
+    } else if (object) {
+      return ActivityObjectHelper.normalizedActorUrl(object as URL);
+    }
+
+    return undefined;
+  }
+
+  public static normalizedToFollow(
+    object: ActivityObject | Link | URL | Array<ActivityObject | URL>
+  ): ActivityObject | Link {
+    if (ActivityObjectHelper.hasProperty(object, "id")) {
+      return { type: "Link", href: (object as ActivityObject).id };
+    } else if (ActivityObjectHelper.hasProperty(object, "name")) {
+      return object as ActivityObject;
+    }
+    return { type: "Link", href: object as URL };
+  }
+
+  public static normalizedObjectFollow(
+    object: ActivityObject | Link | URL | Array<ActivityObject | URL>
+  ): ActivityObject | URL | undefined {
+    if (ActivityObjectHelper.hasProperty(object, "id")) {
+      return (object as ActivityObject).id;
+    } else if (
+      ActivityObjectHelper.hasProperty(object, "type") &&
+      !ActivityObjectHelper.hasProperty(object, "href")
+    ) {
+      return object as ActivityObject;
+    }
+    return object as URL;
+  }
+
+  public static extractId(
+    object: ActivityObject | Link | URL | Array<ActivityObject | URL>
+  ): string | undefined {
+    if (ActivityObjectHelper.hasProperty(object, "id")) {
+      return (object as ActivityObject).id?.toString();
+    } else if (ActivityObjectHelper.hasProperty(object, "name")) {
+      return (object as ActivityObject).name;
+    } else if (ActivityObjectHelper.hasProperty(object, "href")) {
```

Comment:
Be aware that the  url specified under the `href` property does not necessarily needs to be the id of the object. The safe way to get the id through `href` is fetching the `href` url and check the response for an `id` property. This already caused a bug in the backend implementation.